sway-core/src/semantic_analysis/ast_node/expression/typed_expression/if_expression.rs

```diff
@@ -81,7 +71,9 @@ pub(crate) fn instantiate_if_expression(
         .unwrap_or_else(|| type_engine.insert(engines, TypeInfo::Tuple(Vec::new())));
     // if there is a type annotation, then the else branch must exist
     if !else_deterministically_aborts && !then_deterministically_aborts {
-        let (new_warnings, new_errors) = type_engine.unify_with_self(
+        let h = Handler::default();
```

Comment:
Could you add a comment on this usage of `Handler::default()`? This looks different from the majority of the error handler things